catchup/ledgerFetcher.go

```diff
@@ -165,21 +165,9 @@ func (lf *ledgerFetcher) getPeerLedger(ctx context.Context, peer network.HTTPPee
 		if header.Size > maxCatchpointFileChunkSize || header.Size < 1 {
 			return fmt.Errorf("getPeerLedger received a tar header with data size of %d", header.Size)
 		}
-		balancesBlockBytes := make([]byte, header.Size)
-		readComplete := int64(0)
-
-		for readComplete < header.Size {
-			bytesRead, err := tarReader.Read(balancesBlockBytes[readComplete:])
-			readComplete += int64(bytesRead)
-			if err != nil {
-				if err == io.EOF {
-					if readComplete == header.Size {
-						break
-					}
-					err = fmt.Errorf("getPeerLedger received io.EOF while reading from tar file stream prior of reaching chunk size %d / %d", readComplete, header.Size)
-				}
-				return err
-			}
+		balancesBlockBytes, err := io.ReadAll(tarReader)
```

Comment:
if you are not going to `make([]byte, header.Size)` with io.ReadFull and use ReadAll instead, maybe add a manual check that len(balancesBlockBytes) == header.Size? and/or a check that len(balancesBlockSize) or header.Size != 0?

Reply:
You are right. We probably shouldn't trust that the tar reader will not read more bytes than in the header if the file is malformed.